amd_openvx_extensions/amd_rpp/source/tensor/Resample.cpp

```diff
@@ -0,0 +1,251 @@
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+#include "vx_ext_amd.h"
+
+struct ResampleLocalData 
+{
+    vxRppHandle *handle;
+    Rpp32u deviceType;
+    RppPtr_t pSrc;
+    RppPtr_t pDst;
+    float quality;
+    RpptDescPtr pSrcDesc;
+    RpptDescPtr pDstDesc;
+    Rpp32s *sampleFrames;
+    Rpp32s *sampleChannels;
+    Rpp32f *inRateTensor;
+    Rpp32f *outRateTensor;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+};
+
+void update_destination_roi(const vx_reference *parameters, ResampleLocalData *data, RpptROI* src_roi, RpptROI* dst_roi) {
+    float _scale_ratio;
+    for(unsigned i = 0; i < data->pSrcDesc->n; i++) {
+        if (data->inRateTensor[i] != 0)
+            _scale_ratio = data->outRateTensor[i] / (float)data->inRateTensor[i];
+        else
+            _scale_ratio = 0;
+        dst_roi[i].xywhROI.xy.x = (int)std::ceil(_scale_ratio * src_roi[i].xywhROI.xy.x);
+        dst_roi[i].xywhROI.xy.y = src_roi[i].xywhROI.xy.y;
+    }
+}
+
+static vx_status VX_CALLBACK refreshResample(vx_node node, const vx_reference *parameters, vx_uint32 num, ResampleLocalData *data) {
+    vx_status status = VX_SUCCESS;
+    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[5], 0, data->pSrcDesc->n, sizeof(float), data->inRateTensor, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[4], VX_TENSOR_BUFFER_HOST, &data->outRateTensor, sizeof(data->outRateTensor)));
+    void *roi_tensor_ptr_src, *roi_tensor_ptr_dst;
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU)
+    {
+#if ENABLE_OPENCL
+        return VX_ERROR_NOT_IMPLEMENTED;
+#elif ENABLE_HIP
+        return VX_ERROR_NOT_IMPLEMENTED;
+#endif
+    }
+    if (data->deviceType == AGO_TARGET_AFFINITY_CPU)
+    {
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_BUFFER_HOST, &data->pSrc, sizeof(data->pSrc)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_BUFFER_HOST, &data->pDst, sizeof(data->pDst)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_BUFFER_HOST, &roi_tensor_ptr_src, sizeof(roi_tensor_ptr_src)));
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[3], VX_TENSOR_BUFFER_HOST, &roi_tensor_ptr_dst, sizeof(roi_tensor_ptr_dst)));
+
+    }
+    RpptROI* src_roi = reinterpret_cast<RpptROI *>(roi_tensor_ptr_src);
+    RpptROI* dst_roi = reinterpret_cast<RpptROI *>(roi_tensor_ptr_dst);
+    for (uint i = 0; i < data->pSrcDesc->n ; i++) {
+        data->sampleFrames[i] = src_roi[i].xywhROI.xy.x;
+        data->sampleChannels[i] = src_roi[i].xywhROI.xy.y;
+    }
```

Comment:
Would suggest to move this code from L78-L81 to update_destination_roi so that in single loop we can update all the buffers

Reply:
Done